corehq/apps/domain/tests/test_forms.py

```diff
@@ -29,119 +21,239 @@ def test_visible_fields(self):
+        patcher = patch.object(forms, 'domain_has_privilege')
+        self.mock_domain_has_privilege = patcher.start()
+        self.mock_domain_has_privilege.return_value = False
+        self.addCleanup(patcher.stop)
+
+    def _create_form(self):
         return PrivacySecurityForm(user_name='test_user', domain='test_domain')
 
-    def get_visible_fields(self, form):
+    def _get_visible_fields(self, form):
         fieldset = form.helper.layout.fields[0]
         return [field[0] for field in fieldset.fields]
 
 
-class TestDomainGlobalSettingsForm(TestCase):
+class DomainGlobalSettingsFormTests(SimpleTestCase):
+    def test_all_visible_fields(self):
+        form = self._create_form()
+        self.assertEqual([
+            'hr_name',
+            'project_description',
+            'default_timezone',
+            'default_geocoder_location',
+            'logo',
+            'delete_logo',
+            'call_center_enabled',
+            'call_center_type',
+            'call_center_case_owner',
+            'call_center_case_type',
+            'mobile_ucr_sync_interval',
+            'confirmation_link_expiry'
+        ], form.get_visible_field_names())
 
-    def setUp(self) -> None:
-        super().setUp()
-        self.domain = Domain(name='test_domain')
-        self.call_settings = OperatorCallLimitSettings(domain='test_domain')
-        self.call_settings.save()
-        self.domain.save()
-
-    def test_confirmation_link_expiry_not_present_when_flag_not_set(self):
-        set_toggle(TWO_STAGE_USER_PROVISIONING_BY_SMS.slug, self.domain, False, namespace=NAMESPACE_DOMAIN)
-        form = self.create_form()
-        self.assertTrue('confirmation_link_expiry' not in form.fields)
-
-    def test_confirmation_link_expiry_default_present_when_flag_set(self):
-        set_toggle(TWO_STAGE_USER_PROVISIONING_BY_SMS.slug, self.domain, True, namespace=NAMESPACE_DOMAIN)
-        form = self.create_form(confirmation_link_expiry=self.domain.confirmation_link_expiry_time)
-        form.full_clean()
-        form.save(Mock(), self.domain)
-        self.assertTrue('confirmation_link_expiry' in form.fields)
-        self.assertEqual(168, self.domain.confirmation_link_expiry_time)
+    def test_if_cannot_use_custom_logo_logo_fields_are_removed(self):
+        form = self._create_form(can_use_custom_logo=False)
 
-    def test_confirmation_link_expiry_custom_present_when_flag_set(self):
-        set_toggle(TWO_STAGE_USER_PROVISIONING_BY_SMS.slug, self.domain, True, namespace=NAMESPACE_DOMAIN)
-        form = self.create_form(confirmation_link_expiry=100)
-        form.full_clean()
-        form.save(Mock(), self.domain)
-        self.assertTrue('confirmation_link_expiry' in form.fields)
-        self.assertEqual(100, self.domain.confirmation_link_expiry_time)
+        visible_fields = form.get_visible_field_names()
+        self.assertNotIn('logo', visible_fields)
+        self.assertNotIn('delete_logo', visible_fields)
```

Comment:
Should there be another test to verify that these fields are present when `can_use_custom_logo=True`?